try:
    myfile = openfile("test_file.txt", True)
    myfile.writeat("data", 0)
    myfile.close()
    
    try:
        # Open for reading
        f2 = openfile("test_file.txt", False)
        
        # Multiple write attempts - all should fail
        error_count = 0
        
        try:
            f2.writeat("a", 0)
        except:
            error_count = error_count + 1
        
        try:
            f2.writeat("b", 5)
        except:
            error_count = error_count + 1
        
        f2.close()
        
        if error_count != 2:
            log("VULNERABLE: Multiple write attempts allowed\n")
            exitall()
    except Exception, e:
        pass
except Exception, e:
    pass